sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/catalog/v2/ResolveMultipartRelationSuite.scala

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.catalyst.catalog.v2
+
+import org.apache.spark.sql.catalog.v2.{CatalogNotFoundException, CatalogPlugin}
+import org.apache.spark.sql.catalyst.CatalogIdentifier
+import org.apache.spark.sql.catalyst.analysis.{AnalysisTest, Analyzer, EliminateSubqueryAliases, NamedRelation}
+import org.apache.spark.sql.catalyst.expressions.Attribute
+import org.apache.spark.sql.catalyst.parser.CatalystSqlParser
+import org.apache.spark.sql.catalyst.plans.logical.{LeafNode, LogicalPlan}
+import org.apache.spark.sql.catalyst.rules.Rule
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.util.CaseInsensitiveStringMap
+
+private case class TestUnresolvedMultipartRelation(parts: Seq[String])
+    extends LeafNode with NamedRelation {
+
+  override def name: String = "TestUnresolvedMultipartRelation"
+
+  override def output: Seq[Attribute] = Nil
+
+  override lazy val resolved = false
+}
+
+private case class TestMultipartRelation(catalog: Option[CatalogPlugin], ident: CatalogIdentifier)
+    extends LeafNode with NamedRelation {
+
+  override def name: String = "TestMultipartRelation"
+
+  override def output: Seq[Attribute] = Nil
+
+  override lazy val resolved = true
+}
+
+private case class TestMultipartAnalysis(analyzer: Analyzer) extends Rule[LogicalPlan] {
+
+  override def apply(plan: LogicalPlan): LogicalPlan = plan resolveOperators  {
+
+    case TestUnresolvedMultipartRelation(analyzer.CatalogRef(catalog, ident)) =>
+      TestMultipartRelation(catalog, ident)
+  }
+}
+
+private class TestCatalogPlugin(override val name: String) extends CatalogPlugin {
+
+  override def initialize(name: String, options: CaseInsensitiveStringMap): Unit = Unit
+}
+
+class ResolveMultipartRelationSuite extends AnalysisTest {
+  import CatalystSqlParser._
+
+  private val analyzer = makeAnalyzer(caseSensitive = false)
+
+  private val catalogs = Seq("prod", "test").map(name => name -> new TestCatalogPlugin(name)).toMap
```

Comment:
Why not use the plugin system to instantiate these catalogs?

Reply:
It depends on if this test should also be exercising the code path that does said loading, or if we want a separate unit test to verify that behavior in isolation. I'm usually a proponent of each test covering a specific feature and that feature only, so as long as we have another unit test that exercises plugin loading then I think this is fine.

Reply:
To be clear I think we should be avoiding plugin loading in these tests.

Reply:
+1, good call.